core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java

```diff
@@ -180,7 +185,14 @@ private Image getCurrentImageFromCatalog(String currentImageId, CloudbreakImageC
         return imageService.getImage(stack.getId());
     }
 
-    private CloudbreakImageCatalogV3 getImagesFromCatalog(String imageCatalogUrl) throws CloudbreakImageCatalogException {
+    private CloudbreakImageCatalogV3 getImagesFromCatalog(Workspace workspace,
+            com.sequenceiq.cloudbreak.cloud.model.Image image) throws CloudbreakImageCatalogException {
+        String imageCatalogUrl;
+        try {
```

Comment:
Perhaps you could extend these branches with log messages why we are using a certain url.

Reply:
+1